### ISSUE TITLE
PowerDNS does not validate return type of backend data

### ISSUE THREAD

**Issue by cmouse** [PowerDNS/pdns#2624]
When PowerDNS asks for NS record from backend and backend mistakenly returns something else, this is passed unchecked to reply.

**Comment by pieterlexis** [comment]
looks like a dupe of  #512

**Comment by cmouse** [comment]
Not really. This is about, say, pipe or remotebackend, returning crap NS record. It gets copied verbatim into Additional records.
